### PR TITLE
[consolidation] Use proto for communication

### PR DESCRIPTION
**Pull request by jonathanmetzman**
This replaces the current JSON-based system.
This provides the following benefits:
1. Deserialization is improved/simplfied. We reuse ndb's entity->proto code instead of implementing our own hacky deserialization for entities (which didn't support e.g. datetimes) 
2. Type safety. 
3. Typos in parameter names will crash closer to where the mistake is made
4. Speed

### CODE REVIEW

File: src/clusterfuzz/_internal/protos/uworker_msg_pb2.py [hunk 1]
```diff
@@ -0,0 +1,505 @@
+# Copyright 2020 Google LLC
```
Comment (jonathanmetzman) [inline src/clusterfuzz/_internal/protos/uworker_msg_pb2.py:1]: This file is autogenerated and does not need review.

**Comment by Alan32Liu** [comment]
It seems we have many conflits : )

**Comment by jonathanmetzman** [comment]
> It seems we have many conflicts : )

Heh, they were actually quite minor (using uworker_errors.Type.UNHANDLED instead of uworker_msg_pb2.ErrorType.UNHANDLED

**Comment by jonathanmetzman** [comment]
TBR

File: src/clusterfuzz/_internal/bot/tasks/utasks/uworker_io.py [hunk 2]
```diff
@@ -99,28 +84,31 @@ def get_entity_with_properties(ndb_key: ndb.Key, properties) -> ndb.Model:
...
-  serializable = {}
-  ndb_entities = {}
+  uworker_input = uworker_input.copy()
   for key, value in uworker_input.items():
-    if not isinstance(value, ndb.Model):
-      serializable[key] = value
-      continue
-    ndb_entities[key] = make_ndb_entity_input_obj_serializable(value)
+    if isinstance(value, ndb.Model):
+      uworker_input[key] = model._entity_to_protobuf(value)  # pylint: disable=protected-access
```
Comment (Alan32Liu) [inline src/clusterfuzz/_internal/bot/tasks/utasks/uworker_io.py:105]: Would it be better to make the function public, given it has been used publicly in different places?

**Pull request merged**